content/docs/arvo/dill/tasks.md

```diff
@@ -0,0 +1,343 @@
+
+This `task` would not be used from userspace.
+
+#### Returns
+
+Dill returns no `gift` in response to a `%hail` `task`.
+
+## `%heft`
+
+```hoon
+[%heft ~]
+```
+
+Produce memory report.
+
+When Dill receives a `%heft` `task` it passes Arvo a `%whey` `waif` to obtain a memory report and prints it to the terminal.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%heft` `task`.
+
+## `%hook`
+
+```hoon
+[%hook ~]
+```
+
+This terminal hung up.
+
+This task is not used.
+
+## `%harm`
+
+```hoon
+[%harm ~]
+```
+
+All terminals hung up.
+
+This `task` is not used.
+
+## `%init`
+
+```hoon
+[%init ~]
+```
+
+This `task` is called only once, when Arvo first enters the [adult stage](/docs/arvo/overview#structural-interface-core). It performs initial setup for Dill, such as setting the width of the console.
+
+Note that this is not actually the first `task` passed to Dill - see [%boot](#%boot).
+
+This `task` would not be used from userspace.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%init` `task`.
+
+## `%meld`
+
+```
+[%meld ~]
+```
+
+Deduplicate persistent state.
+
+Dill asks the runtime to perform the memory deduplication.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%meld` `task`.
+
+## `%noop`
+
+```hoon
+[%noop ~]
+```
+
+No operation.
+
+A `%noop` `task` does nothing, as the name implies.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%noop` `task`.
+
+## `%pack`
+
+```hoon
+[%pack ~]
+```
+
+Defragment persistent state.
+
+Dill asks the runtime to perform the defragmentation.
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%meld` `task`.
+
+## `%talk`
+
+```hoon
+[%talk p=tank]
+```
+
+This `task` is not used.
+
+## `%text`
+
+```hoon
+[%text p=tape]
+```
+
+Print `tape` to terminal.
+
+Upon receiving a `%text` `task`, Dill will convert the `tape` given in `p` to a `(list @c)` and give it to the runtime in a `%blit` `gift` including a `%lin` [$blit](/docs/arvo/dill/data-types#blit).
+
+#### Returns
+
+Dill does not return a `gift` in response to a `%text` `task`, but it does give a `%blit` `gift` to the runtime which looks like:
+
+```hoon
+[%blit p=(list blit)]
+```
+
+## `%view`
+
+```hoon
+[%view session=~]
+```
+
+Watch session.
+
+A `%view` `task` subscribes for a copy of all `%blit`s which Dill sends to the Unix terminal. This `task` is used by the `%herm` app so it can convert the `$blit`s to JSON and mirror the Unix terminal in the web terminal.
+
+The ability to specify a session is not yet implemented in Dill, so `session` is always `~` and it's always the default session.
+
+#### Returns
+
+Dill will `%give` a copy of all `%blit`s it sends to the Unix terminal. A `%blit` `gift` is:
+
+```hoon
+[%blit p=(list blit)]
+```
+
+## `%trim`
+
+```hoon
+[%trim p=@ud]
+```
+
+`%trim` is a common vane `task` used to reduce memory usage. However, it does
+nothing for Dill.
```

Comment:
May or may not be worth mentioning this is because dill keeps only very minimal and necessary state.